src/main/webapp/user_profile_script.js

```diff
@@ -80,22 +78,56 @@ async function abandonTask(keyString) {
     }
 }
 
+async function verifyTask(keyString) {
```

Comment:
I think it would be better to combine these two into an `updateTask(Keystring, status)` function, since they are very similar - the `completeTask` and `abandonTask` functions could be combined as well, but this is just my personal preference!

Reply:
Personally I don't think it's a good idea to combine the two functions because the alert messages are different. So even if I combine the two functions, I need extra if-else statements to send out different alert messages depending on the input `status`, which will also create a lot of code duplication. And for `completeTask` and `abandonTask`, their behavior is different. So I think I won't combine those two functions.

Reply:
Sounds good :) !